mlrun/runtimes/function.py

```diff
@@ -45,6 +47,53 @@
 default_max_replicas = 4
 
 
+def validate_nuclio_version_compatibility(*min_versions):
+    """
+    Validation is best effort - if we can't parse version we assume compatible.
+    This method mainly used via decorator 'min_nuclio_versions'.
+    :param min_versions: valid version(s), assuming no 2 versions has equal major and minor.
+    """
+    parsed_min_versions = [
+        semver.VersionInfo.parse(min_version) for min_version in min_versions
+    ]
+    try:
+        parsed_current_version = semver.VersionInfo.parse(mlconf.nuclio_version)
+    except ValueError:
+        logger.warning(
+            "Unable to parse nuclio version, assuming compatibility",
+            nuclio_version=mlconf.nuclio_version,
+        )
+        return True
+    compatible = False
+    for parsed_min_version in parsed_min_versions:
+        if (
+            parsed_current_version.major == parsed_min_version.major
+            and parsed_current_version.minor == parsed_min_version.minor
+            and parsed_current_version.patch < parsed_min_version.patch
+        ):
+            return False
+        if parsed_current_version >= parsed_min_version:
+            compatible = True
```

Comment:
You can already return True here, no need to keep iterating

Reply:
Actually if we return True immediately than consider `with_node_selection` that is available since nuclio 1.5.20 and 1.6.10 if I have nuclio 1.6.9 I will return True when checking if 1.6.9 >= 1.5.20 but I want to return False, I agree it is unclear from the code I will think of a comment that explains it

Reply:
discussed offline, using sort solves this and makes the flow clearer